main/design_documents/ARM_NEON_SVE_BRIDGE.rst

```diff
@@ -0,0 +1,44 @@
+=======================================
+Design Document for ARM_NEON_SVE_BRIDGE
+=======================================
+
+This document describes the design of the intrinsics defined in the ("NEON_SVE bridge")
+<../acle.rst#_ssec-NEONSVEBridge>. These intrinics add a
+convienient way to use NEON and SVE vectors interchangeably in C code with
+no performance overhead.
+
+Location
+########
+
+The intrinsics will be in a new header file called ``arm_extensions_bridge.h``.
+
+They will be conditionally defined with
+``#define __ARM_NEON_SVE_BRIDGE (__ARM_FEATURE_NEON && __ARM_FEATURE_SVE)``.
+``__ARM_FEATURE_NEON`` is used as it is aarch64 specific, and SVE is not
+supported in the aarch32 state.
+
+Including this header file will define ``__ARM_NEON_SVE_BRIDGE`` as 1.
+
+Suggestions & Considerations
+############################
+
+Why not enable the extraction of any 128-bit subvector in an SVE vector?
+------------------------------------------------------------------------
+
+Extracting other 128-bit subvectors would go against the no-op design.
+There are currently other methods of getting data this way from SVE
+subvectors using SVE intrinsics.
```

Comment:
Could you add an example of how to extract another 128-bit segment from an SVE vector? I am assuming the ideal codegen here would be `MOV <Zd>.<T>, <Zn>.<T>[<imm>]` but can't think of a sequence that would produce this off the top of my head?

Reply:
`svdupq_lane_s8(x, 1);` compiles to `mov     z0.q, z0.q[1]` on the head of llvm